### PR TITLE
[ONNX] Enable aten:normal op and add tests for aten:uniform op.

### PR DESCRIPTION
**Pull request by fatcat-z**
1. Add a new symbolic function broadcast_tensors() to support exporting torch.broadcast_tensors() function. This is required by exporting torch.distribution.normal() function.
2. Add a new symbolic function normal() to support exporting torch.distribution.normal() function.
3. Add relative tests for normal and uniform ops as well.

### CODE REVIEW

File: torch/onnx/symbolic_opset11.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#!/usr/bin/env python3
```
Comment (neginraoof) [inline torch/onnx/symbolic_opset11.py:1]: Are these two lines needed?

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 2]
```diff
@@ -9246,6 +9246,103 @@ def forward(self, x):
...
+                                   training=torch.onnx.TrainingMode.EVAL)
+
+        ort_out = []
+        for i in range(1500):
+            ort_out.append(run_ort(ort_sess, input=dummy_input))
+
+        actual_var = np.var(ort_out)
+        actual_mean = np.mean(ort_out)
+
+        assert abs(abs(actual_mean) - expected_mean) <= expected_mean * 0.15, \
```
Comment (neginraoof) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:9305]: Why are you using the 0.15 constant?

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 3]
```diff
@@ -9246,6 +9246,103 @@ def forward(self, x):
...
+        model_export = M()
+        dummy_input = (torch.tensor([expected_min]), torch.tensor([expected_max]))
+        ort_sess = convert_to_onnx(model_export, input=dummy_input, opset_version=self.opset_version,
+                                   training=torch.onnx.TrainingMode.EVAL)
+
+        ort_out = []
+        for i in range(500):
+            ort_out.append(run_ort(ort_sess, input=dummy_input))
+
+        actual_min = min(ort_out)
```
Comment (neginraoof) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:9340]: Mean of a uniform random variable = 1/2(a+b)
Is it possible to compare the actual mean with expected mean?
Also, expected var =  (b − a)*2/12.

**Review by neginraoof — COMMENTED** [review]
Thanks, looks good overall. Had some minor comments.

File: test/onnx/test_pytorch_onnx_onnxruntime.py (hunk 2 above)
Comment (fatcat-z) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:9305]: After some tests, 0.15 is a stable offset between our PyTorch value and ONNX value for many rounds of running this test. And I also think it is an acceptable one for the accuracy validation.

File: torch/onnx/symbolic_opset11.py (hunk 1 above)
Comment (fatcat-z) [reply torch/onnx/symbolic_opset11.py:1]: In my comments of normal() method, there are some special characters like 'σ' and 'μ' which was used to explain why we implement this function in this way.

I believe these characters caused a new flake8 issue(encoding) which is resolved by including these 2 lines.

File: test/onnx/test_pytorch_onnx_onnxruntime.py (hunk 3 above)
Comment (fatcat-z) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:9340]: No, comparing mean value in this scenario is not right. Uniform() method has a limitation on min and max value, so we should compare them directly if no blocker.

Let's say if we pass min and max value to Uniform() as below:
min: 5
max: 10
Then the mean value should be 7.5((5 + 10)/2).

But if the min and max value are set to below, the mean value could also be 7.5 but the generated values might be out of the above given scope and this should not be acceptable:
min: 6
max: 9

File: torch/onnx/symbolic_opset11.py [hunk 4]
```diff
@@ -994,3 +996,30 @@ def repeat_interleave(g, self, repeats, dim=None, output_size=None):
...
+
+def normal(g, loc, scale, seed):
+    # If you can sample from a given distribution with mean 0 and variance 1, then you can easily sample from a
+    # scale-location transformation of that distribution, which has mean μ and variance, σ's square. If x is a sample
+    # from a mean 0 and variance 1 distribution then
+    #       σx+μ
+    # is a sample with mean μ and variance σ's square.
+    result = zeros_like(g, loc)
+    normal_base = add(g, result, g.op("RandomNormal", shape_i=[1]))
+    result = sqrt(g, scale)
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset11.py:1023]: scale is standard variance, no need to call sqrt

File: test/onnx/test_pytorch_onnx_onnxruntime.py (hunk 2 above)
Comment (BowenBao) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:9305]: The expected mean also should follow the normal distribution, with (mean, variance/sample_size). Using the interval calculator for normal distribution https://keisan.casio.com/exec/system/14060744929691, the probability of passing the test is only 99.6%. On average it takes ~250 runs to produce a failure case. One suggestion would be to increase the sample size from 1500 to 10000, which guarantees 0.99999999999994 success rate. 

np.var is computing variance, however variable `expected_var` should actually be `expected_std`, since that's the input `torch.distributions.Normal` take. And with the redundant sqrt in the symbolic, although the exported graph is not the same with pytorch, the test assertion actually passed... For sample variance, it is following chi square distribution with degree of freedom of sample_size - 1. Using sample size of 10000 should also provide good success rate for the check.

And to increase performance of the test, it is recommended to generate 10000 samples directly in the model `torch.distributions.Normal(x, y).sample([10000])`. This should still be fast enough.

**Review by BowenBao — CHANGES_REQUESTED** [review]

File: torch/onnx/symbolic_opset11.py [hunk 5]
```diff
@@ -994,3 +996,30 @@ def repeat_interleave(g, self, repeats, dim=None, output_size=None):
...
+def broadcast_tensors(g, self):
+    all_tensors = sym_help._unpack_list(self)
+    final_result = zeros_like(g, all_tensors[0])
+
+    for t in all_tensors:
+        final_result = add(g, final_result, t)
+
+    outputs = g.op("SequenceEmpty", dtype_i=sym_help.cast_pytorch_to_onnx["Float"])
+    for t in all_tensors:
+        outputs = g.op("SequenceInsert", outputs, expand_as(g, t, final_result))
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset11.py:1010]: use single `g.op("SequenceConstruct")`

File: torch/onnx/symbolic_opset11.py [hunk 6]
```diff
@@ -994,3 +996,30 @@ def repeat_interleave(g, self, repeats, dim=None, output_size=None):
...
+
+    return outputs
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset11.py:1012]: curious how is broadcast_tensors consumed? i.e. what node takes the output of it as inputs

Comment (fatcat-z) [reply torch/onnx/symbolic_opset11.py]: Here is an example of ATEN graph by one of test:

=== ATEN graph:
graph(%0 : Float(1, strides=[1], requires_grad=0, device=cpu),
      %1 : Float(1, strides=[1], requires_grad=0, device=cpu)):
  %6 : Tensor[] = prim::ListConstruct(%0, %1)
  %7 : Tensor[] = aten::broadcast_tensors(%6) # /home/jay/repos/fatcat-z/pytorch/torch/functional.py:72:0
  %x : Float(1, strides=[1], requires_grad=0, device=cpu), %y : Float(1, strides=[1], requires_grad=0, device=cpu) = prim::ListUnpack(%7)
  %30 : Long(requires_grad=0, device=cpu) = prim::Constant[value={0}]()
  %11 : Long(device=cpu) = aten::size(%x, %30) # /home/jay/repos/fatcat-z/pytorch/torch/distributions/normal.py:49:0
  %31 : Long(requires_grad=0, device=cpu) = prim::Constant[value={20000}]()
  %20 : int[] = prim::ListConstruct(%31, %11)
  %32 : Bool(requires_grad=0, device=cpu) = prim::Constant[value={0}]()
  %22 : Float(20000, 1, strides=[0, 1], requires_grad=0, device=cpu) = aten::expand(%x, %20, %32) # /home/jay/repos/fatcat-z/pytorch/torch/distributions/normal.py:64:0
  %33 : Long(requires_grad=0, device=cpu) = prim::Constant[value={20000}]()
  %25 : int[] = prim::ListConstruct(%33, %11)
  %34 : Bool(requires_grad=0, device=cpu) = prim::Constant[value={0}]()
  %27 : Float(20000, 1, strides=[0, 1], requires_grad=0, device=cpu) = aten::expand(%y, %25, %34) # /home/jay/repos/fatcat-z/pytorch/torch/distributions/normal.py:64:0
  %28 : NoneType = prim::Constant()
  %29 : Float(20000, 1, strides=[1, 1], requires_grad=0, device=cpu) = aten::normal(%22, %27, %28) # /home/jay/repos/fatcat-z/pytorch/torch/distributions/normal.py:64:0
  return (%29)

Comment (BowenBao) [reply torch/onnx/symbolic_opset11.py]: i see, i think it would be better to export as `prim::ListConstruct` at the end. If this is possible, the symbolic function can be moved to symbolic_opset9.py and work for all opsets.

**Review by BowenBao — APPROVED** [review]
